### PR TITLE
Div 3747 amend petition exclude reason

### PR DESCRIPTION
**Pull request by mounikahmcts**
# Description
As a Petitioner
I want to be presented only with Facts that I can actually use on my amended petition
so that I do not submit my amended application using a previously rejected reason.

Fixes # https://tools.hmcts.net/jira/browse/DIV-3747

### CODE REVIEW

File: app/steps/grounds-for-divorce/reason/index.js [hunk 1]
```diff
@@ -87,6 +96,29 @@ module.exports = class ReasonForDivorce extends ValidationStep {
 
       const timeSinceMarriage = moment().diff(marriageDate, 'years');
 
+      if (ignoreDivorceReasons && ignoreDivorceReasons.length > 0) {
```
Comment (dtaylor7) [inline app/steps/grounds-for-divorce/reason/index.js]: perhaps a little comment above might help future devs i.e. `remove all previous attempted reasons for divorce`

Comment (mounikahmcts) [reply app/steps/grounds-for-divorce/reason/index.js]: Added comment

**Comment by mounikahmcts** [comment]
> 👍
> Any blockers on this? It has been open for 1+ month

The reason is it cannot be tested before amend petition backend stuff is ready in place.

**Comment by dtaylor7** [comment]
Changes merged into https://github.com/hmcts/div-petitioner-frontend/pull/386

**Pull request closed without merging**